src/limitedmap.h

```diff
@@ -66,8 +66,7 @@ class limitedmap
     }
     void update(const_iterator itIn, const mapped_type& v)
     {
-        // TODO: When we switch to C++11, use map.erase(itIn, itIn) to get the non-const iterator.
-        iterator itTarget = map.find(itIn->first);
+        iterator itTarget = map.erase(itIn, itIn);
```

Comment:
I currently don't have access to my laptop, so will bench and fix it
accordingly in a couple of days.

On Aug 3, 2016 7:25 PM, "Wladimir J. van der Laan" <notifications@github.com>
wrote:

> In src/limitedmap.h
> <https://github.com/bitcoin/bitcoin/pull/8353#discussion_r73304022>:
>
> > @@ -66,8 +66,7 @@ class limitedmap
> >      }
> >      void update(const_iterator itIn, const mapped_type& v)
> >      {
> > -        // TODO: When we switch to C++11, use map.erase(itIn, itIn) to get the non-const iterator.
> > -        iterator itTarget = map.find(itIn->first);
> > +        iterator itTarget = map.erase(itIn, itIn);
>
> I strongly prefer non-tricky code to tricky code, even if documented, to
> be honest. It increases the scope of making mistakes (due to
> misunderstanding the code) in later maintenance.
> So unless this makes a clear performance difference in practical usage I
> prefer keeping it as-is.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/8353/files/3debfbca0d85d7c17f67c5a15df0938584445026#r73304022>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHSu6XP5fp8pV6uoGcv3MUlHEtEF-t1Nks5qcFmmgaJpZM4JOU7Z>
> .
>